tensorflow/compiler/xla/stream_executor/blas.h

```diff
@@ -186,6 +187,82 @@ class AlgorithmConfig {
 typedef int64_t ComputePrecision;
 constexpr ComputePrecision kDefaultComputePrecision = 0;
 
+// Call context information for GEMM API calls
+// This is extra information that can optionally be passed down to the blas
+// library, so that it can pick the efficient imlpementation based on context
+enum class CallContext {
+  kNone = 0,            // No information
+  kForward = 1,         // call happens in "forward" pass
+  kBackpropInput1 = 2,  // call happens in "backprop" pass for the first input
+  kBackpropInput2 = 4,  // call happens in "backprop" pass for the second input
+};
+
+struct GemmCall {
+  blas::Transpose transa, transb;
+  uint64_t m; 
+  uint64_t n;
+  uint64_t k;
+  DataType dtype;
+  // At the Stream interface and above, same type as pa/pb/c
+  // Inside SE, ditto except it's float when pa/pb/c are half or bf16
+  const void *alpha;
```

Comment:
Can we do a little better than a void pointer to data owned by something else? Maybe copy the data into a variant or something?